app/build.gradle

```diff
@@ -2,6 +2,7 @@ plugins {
     id 'com.android.application'
```

Comment:
No has inicializado firebase en el Application . Te falta mucho desarrollo.

Iniciar https://firebase.google.com/docs/android/setup?hl=es
Para cuando lo tengas montado: https://www.youtube.com/watch?v=KYPc7CAYJOw

import com.google.firebase.FirebaseApp;

public class MyApplication extends Application {
    @Override
    public void onCreate() {
        super.onCreate();
        // Inicializa Firebase
        FirebaseApp.initializeApp(this);
    }
}